tools/Jenkins/dependency_spotter.py

```
@@ -191,15 +191,17 @@ def extract_package_versions(url: str, os_name: str) -> Dict[str, str]:
     package_version_dict = {}
-    with urllib.request.urlopen(url) as file:
-        for line in file.readlines():
-            regex_result = re.search(pattern=regex_pattern, string=line.decode("utf-8"))
-            if regex_result is not None and len(regex_result.groups()) == 3:
-                package_name = regex_result.group(1)
-                if package_name.startswith("mantid"):
-                    continue
-                version = regex_result.group(2)
-                package_version_dict[package_name] = version
+    if not url.startswith(("http:", "https:")):
+        raise ValueError("URL must start with 'http:' or 'https:'")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)